### PR TITLE
[SPARK-26319][SQL][Test] Add appendReadColumns Unit Test for HiveShimSuite

### PR DESCRIPTION
**Pull request by da-liii**
## What changes were proposed in this pull request?

Add appendReadColumns Unit Test for HiveShimSuite.

## How was this patch tested?
```
$ build/sbt
> project hive
> testOnly *HiveShimSuite
```

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala [hunk 1]
```diff
@@ -53,19 +53,12 @@ private[hive] object HiveShim {
...
-        result.append(',')
-      }
-      result.append(col)
-    }
-    conf.set(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, result.toString)
+    val key = ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR
+    val value = Option(conf.get(key, null))
+      .map(old => cols.+:(old))
+      .getOrElse(cols)
+      .mkString(",")
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: I don't think this is more readable. Also, the previous code is more performant. I wouldn't change this.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala [hunk 2]
```diff
@@ -53,19 +53,12 @@ private[hive] object HiveShim {
...
-        first = false
-      } else {
-        result.append(',')
-      }
-      result.append(col)
-    }
-    conf.set(ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR, result.toString)
+    val key = ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR
+    val value = Option(conf.get(key, null))
+      .map(old => cols.+:(old))
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: Also, looks the output would be different after this change.
Looks it was `old + col` but the current changes looks doing `col + old`

**Comment by HyukjinKwon** [comment]
Let's close this one.

**Comment by gatorsmile** [comment]
@sadhen What is the motivation of this PR?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 1 above)
Comment (sadhen) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: As for performance:

Current Version:
```
[info] HiveShimSuite:
[info] - appendReadColumns (549 milliseconds)
```

Previous Version:
```
[info] HiveShimSuite:
[info] - appendReadColumns (949 milliseconds)
```

Comment (sadhen) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: For comprehension is just a syntax sugar and is not performant at all.

**Comment by sadhen** [comment]
Nevermind, just do not like the coding style personally.

**Comment by gatorsmile** [comment]
The existing way is too JAVA.

**Comment by HyukjinKwon** [comment]
Yup, the test looks okay in that way. Let's file a JIRA and only leave the test case.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 1 above)
Comment (HyukjinKwon) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: Right, but I don't think it's more readable. It's non-critical path so performance concern is secondary anyway.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 2 above)
Comment (HyukjinKwon) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: ? output is different, no?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 1 above)
Comment (maomaoChibei) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: thanks HyukjinKwon for the kind explaination. we are from an alibaba subsidiary internet company working on resolving real time data calculation based on Peta level data. Currently we have an real time engine which is built on this spark-sql, well, at lease from the sql engine point of view. The throughput is over an qps of 1k and with response latency less than 100ms. Its an none-stop online 24*7 platform serving over ten million active users. If this market scale and platform scale meets the criteria, let me know, we are searching for further cooporations. thanks again. maomao

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 2 above)
Comment (HyukjinKwon) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: Ah, it's `:+` not `+:`. Yea, it's confusing.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 1 above)
Comment (HyukjinKwon) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: Thanks. It doesn't matter which company it is. All the PRs are equally and reasonably reviewed, and merged per the same criteria.

**Comment by sadhen** [comment]
OK, I will modify the PR several hours later.

**Comment by sadhen** [comment]
I've revert the refactor commit.

I'm wondering if I need to create a issue for a unit test only PR.

**Comment by sadhen** [comment]
@HyukjinKwon  Please re-review.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala (hunk 1 above)
Comment (maomaoChibei) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala]: gotcha

**Review by HyukjinKwon — APPROVED** [review]
Looks fine.

**Comment by HyukjinKwon** [comment]
Let's fix PR description as well. You can leave the comments above resolved.

**Comment by sadhen** [comment]
@HyukjinKwon I've updated the desc.

**Pull request closed without merging**